server/src/workspace/workspace-schema-storage/workspace-schema-storage.service.ts

```diff
@@ -16,8 +14,32 @@ export class WorkspaceSchemaStorageService {
     >,
     @InjectMemoryStorage('typeDefs')
     private readonly typeDefsMemoryStorageService: MemoryStorageService<string>,
+    @InjectMemoryStorage('cacheVersion')
+    private readonly cacheVersionMemoryStorageService: MemoryStorageService<string>,
+    private readonly workspaceCacheVersionService: WorkspaceCacheVersionService,
   ) {}
 
+  async validateCacheVersion(workspaceId: string): Promise<void> {
+    const currentVersion =
+      (await this.cacheVersionMemoryStorageService.read({
+        key: workspaceId,
+      })) ?? '0';
+    const latestVersion = await this.workspaceCacheVersionService.getVersion(
+      workspaceId,
+    );
+
+    if (currentVersion !== latestVersion) {
+      // Invalidate cache if version mismatch is detected
+      await this.invalidateCache(workspaceId);
+
+      // Update the cache version after invalidation
+      await this.cacheVersionMemoryStorageService.write({
```

Comment:
Should we update the version when we actually write the new data in the cache instead?